### ISSUE TITLE
No error on reassigning Final vars in class bodies

### ISSUE THREAD

**Issue by jfmend** [microsoft/pyright#7797]
## Environment data
Tested on:
- pylance v2024.4.1 
- python 3.10.12

## Code Snippet

```python
from typing import Final
class A:
    n: Final = 2
    n = 3
```

The type hint for `n` is `(constant) n: Literal[2]` before reassigning and `(constant) n: Literal[3]` afterwards.

## Expected behavior

An error message at the reassigning of the constant `n`.

## Actual behavior

No error message.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release of pyright.

**Comment by jfmend** [comment]
Thank you for the quick fix and for your great work!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.361, which I just published.
